compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5033,6 +5039,19 @@ public void visit(BLangXMLElementAccess xmlElementAccess) {
         result = rewriteExpr(invocationNode);
     }
 
+    private BLangInvocation generateRequireTypeLangLibFunction(BLangAccessExpression fieldAccessExpr) {
+        ArrayList<BLangExpression> requiredArgs = new ArrayList<>();
+        BType typedescType = new BTypedescType(fieldAccessExpr.type, null);
+        BLangTypedescExpr typedescExpr = new BLangTypedescExpr();
+        typedescExpr.resolvedType = fieldAccessExpr.type;
+        typedescExpr.type = typedescType;
+        requiredArgs.add(typedescExpr);
+        fieldAccessExpr.desugared = true;
```

Comment:
What is the purpose of enabling the desugared field?

Reply:
I did this to stop desugaring `fieldAccessExpr` repeatedly(this happen due to use of `fieldAccessExpr` as a parameter to the `requiredType` function ). Anyway, I removed this function with the new changes.